### PR TITLE
Allowing tool specified env passing

### PR DESCRIPTION
**Pull request by bgruening**
This will allow a tool developer to specify which environment variables should be passed into the container. This will bring us closer to get Jupyter, RStudio etc. working.

Currently, only htcondor is supported.

### CODE REVIEW

File: test/functional/tools/realtimetool_juypter_notebook.xml [hunk 1]
```diff
@@ -5,7 +5,10 @@
     <realtime>
         <entry_point name="Jypter Interactive Environment">
             <port>8888</port>
-            <url>ipython/tree</url>
+            <url>$mode.notebook_entry_point</url>
+            <api_key>true</api_key>
+            <history_id>true</history_id>
+            <infrastructure_url>true</infrastructure_url>
```
Comment (blankenberg) [inline test/functional/tools/realtimetool_juypter_notebook.xml:11]: I feel like these should not be associated at the individual entry point level. Seem more like overall tool/job level attributes -- outside of the `<realtime>` tagset.

Comment (bgruening) [reply test/functional/tools/realtimetool_juypter_notebook.xml:11]: Outside of the `<realtime>`? Maybe outside of `entry_point` but it belongs to the `<realtime>` tool isn't it? api_key is not useful outside of the realtime setup, isn't it?

Comment (jmchilton) [reply test/functional/tools/realtimetool_juypter_notebook.xml:11]: Yeah... lets just always inject the history id and infrastructure URL for all "entry points" containing tools. The API key thing should be somewhere else and available to regular tools right I guess - if the conceit is we want these to be normal tools.

**Comment by blankenberg** [comment]
@bgruening @jmchilton should we close this?

**Comment by bgruening** [comment]
I think so ...

**Pull request closed without merging**